crates/sui/src/keytool.rs

```diff
@@ -339,36 +339,37 @@ impl KeyToolCommand {
             }
 
             KeyToolCommand::Import {
-                mnemonic_phrase,
+                input_string,
                 key_scheme,
                 derivation_path,
             } => {
-                let address =
-                    keystore.import_from_mnemonic(&mnemonic_phrase, key_scheme, derivation_path)?;
-                info!("Key imported for address [{address}]");
+                // check if it is a private key or a mnemonic
+                if input_string.starts_with("0x") && input_string.len() == 66 {
+                    let base64 = convert_string_to_base64(input_string)?;
+                    let skp = SuiKeyPair::decode_base64(&base64);
+                    if let Ok(skp) = skp {
+                        keystore.add_key(skp)?;
+                        println!("Private key imported successfully.")
+                    } else {
+                        return Err(anyhow!(
+                            "Cannot decode base64 from private key. Importing failed."
+                        ));
+                    }
+                } else if input_string.starts_with("0x") && input_string.len() != 66 {
+                    return Err(anyhow!(
+                        "Private key is malformed. Expected key length of 66 but got {}",
+                        input_string.len()
+                    ));
+                } else {
+                    keystore.import_from_mnemonic(&input_string, key_scheme, derivation_path)?;
+                    println!("Mnemonic imported successfully.")
+                }
```

Comment:
nit: I have a slight preference for an early return style -- I don't think this code is overly complex, but when we start adding conditions (especially error conditions, as we probably will need to add to handle the key_scheme and derivation_path fields), I find that it's easier to manage in Rust this way:

```rust
                if input_string.starts_with("0x") {
                    if input_string.len() != 66 {
                        return Err(anyhow!(
                            "Private key is malformed. Expected key length of 66 but got {}",
                            input_string.len()
                        ));
                    }

                    let base64 = convert_string_to_base64(input_string)?;
                    let skp = SuiKeyPair::decode_base64(&base64);
                    
                    let Ok(skp) = SuiKeyPair::decode_base64(&base64) else {
                        return Err(anyhow!(
                            "Cannot decode base64 from private key. Importing failed."
                        ));
                    }

                    keystore.add_key(skp)?;
                    eprintln!("Private key imported successfully.")
                } else {
                    keystore.import_from_mnemonic(&input_string, key_scheme, derivation_path)?;
                    eprintln!("Mnemonic imported successfully.")
                }
```

Also, nit: using `stderr` for the informational messages.

Reply:
That looks a lot more neat, fixed it.